cidc_api/services/ingestion.py

```diff
@@ -136,7 +136,22 @@ def upload_manifest():
     """
     schema_hint, schema_path, xlsx_file = extract_schema_and_xlsx()
 
-    return NotImplemented("Manifest ingestion is not yet supported.")
+    md_patch, file_infos = prism.prismify(xlsx_file, schema_path, schema_hint)
+
+    if len(file_infos) > 0:
+        raise BadRequest(f"Unexpected local files in template for {schema_hint}")
+
+    try:
+        trial_id = md_patch[TRIAL_ID_FIELD]
+    except KeyError:
+        raise BadRequest(f"No {TRIAL_ID_FIELD} found/parsed from template.")
+
+    try:
+        TrialMetadata.patch_manifest(trial_id, md_patch)
+    except AssertionError as e:
+        raise BadRequest(f"Trial with {TRIAL_ID_FIELD} {trial_id} not found.")
```

Comment:
Can we be totally certain nothing else in `TrialMetadata.patch_manifest` will throw an assertion error?